server/Knowledge_checklist.sql

```diff
@@ -0,0 +1,100 @@
+drop table if exists students cascade;
+drop table if exists mentors cascade;
+drop table if exists region;
+drop table if exists learning_objectives cascade;
+drop table if exists mapping_skills cascade;
+drop table if exists tech_skills cascade;
+drop table if exists competency_level cascade;
+
+CREATE TABLE students (
+    id          SERIAL PRIMARY KEY,
+    name        VARCHAR (50)NOT NULL,
+    location    VARCHAR (50)NOT NULL,
+    email       VARCHAR (150) NOT NULL,
+    password    VARCHAR(70) NOT NULL
+);
+
+CREATE TABLE mentors (
+    id          SERIAL PRIMARY KEY,
+    name        VARCHAR (50)NOT NULL,
+    location    VARCHAR (50)NOT NULL,
+    email       VARCHAR (150) NOT NULL,
+    password    VARCHAR(70) NOT NULL
+);
+
+CREATE TABLE region (
+    id          SERIAL PRIMARY KEY,
+    location    VARCHAR(50) NOT NULL 
+);
+
+CREATE TABLE learning_objectives (
+    id           SERIAL PRIMARY KEY,
+    obj_id       VARCHAR (50)NOT NULL,
+    lesson_id    INT NOT NULL            
+);
+
+CREATE TABLE mapping_skills (
+    id          SERIAL PRIMARY KEY,
+    stud_id     INT REFERENCES students(id), 
+    obj_id      INT NOT NULL,    
+    comp_id     INT NOT NULL
+         
+);
+
+CREATE TABLE tech_skills (
+    id          SERIAL PRIMARY KEY,
+    tech_id     VARCHAR(50)NOT NULL,
+    stud_id     INT NOT NULL,
```

Comment:
There is already the `mapping_skills` table to link students with objectives and competency levels, I don't think the `tech_skills` table needs to store foreign keys.

Be careful that if table A has a relation to table B, and table B has a relation to table A, it can lead to data inconsistency. It is enough for table A to reference table B.

Reply:
AO amended